index.bs

```diff
@@ -1627,6 +1664,39 @@ Releases the resources associated with the {{MLTensor}}. This method is idempote
 
 Note: Since no further operations can be enqueued using this tensor, implementations can free any additional resource allocations associated with this tensor once all previously submitted operations using it are complete.
 
+## {{MLConstantTensor}} interface ## {#api-mlconstanttensor}
+
+The {{MLConstantTensor}} interface represents a tensor which may be used as a constant to an {{MLGraph}}. The memory backing an {{MLConstantTensor}} should be allocated in an [=implementation-defined=] fashion according to the requirements of the {{MLContext}} and the {{MLOperandDescriptor}} used to create it. Operations involving the {{MLTensor/[[data]]}} of an {{MLConstantTensor}} occur on the {{MLContext/[[timeline]]}} of its associated {{MLContext}}.
+
+The [=implementation-defined=] requirements of how an {{MLConstantTensor}} is allocated may include constraints such as that the memory is allocated with a particular byte alignment or in a particular memory pool.
+
+<script type=idl>
+[SecureContext, Exposed=(Window, Worker)]
+interface MLConstantTensor : MLTensor {
+};
+</script>
+
+### Creating an {{MLConstantTensor}} ### {#api-mlconstanttensor-create}
+
+An {{MLConstantTensor}} is created by its associated {{MLContext}}.
+
+<details open algorithm>
+  <summary>
+    To <dfn>create an MLConstantTensor</dfn> given {{MLContext}} |context|, {{MLOperandDescriptor}} |inputDescriptor|, run the following steps:
+  </summary>
+    1. Let |realm| be |context|'s [=relevant realm=].
+    1. Let |tensor| be a new {{MLConstantTensor}} in |realm|.
+    1. Set |tensor|.{{MLTensor/[[context]]}} to |context|.
+    1. Let |descriptor| be a new {{MLTensorDescriptor}}.
```

Comment:
There's nothing wrong with this, but the lint tool will flag subsequent lines because it wants a variable named `descriptor` to be an `MLOperandDescriptor` in subsequent steps. (Without that lint rule it's easy to link to the wrong thing.) If you rename this `tensorDescriptor` the lint tool will not complain.

Reply:
Also, a different issue: the lint tool will flag this line with `Let ... be a new ...` as requiring a "realm". This is a false positive since dictionaries are just maps and don't need a realm. To quiet that, we'll need to update the lint tool here:

https://github.com/webmachinelearning/webnn/blob/4c8c75e200f8930f30d1b10c8dbecb8e6c953eeb/tools/lint.mjs#L409

Just add `"MLTensorDescriptor"` to the list of dictionary types. Sorry about that! (I have a to-do item to parse the spec's IDL and determine what is a dictionary automagically, but that hasn't happened yet.)

Reply:
@inexorabletash Do I revert? Looks like lint.mjs has been updated.

Reply:
Yes! Sorry about that... I wasn't sure if I could get a more intelligent change in; it went much more quickly than I expected! @bbernhar